clang/runtime/dpct-rt/include/dpct/graph.hpp

```diff
@@ -25,6 +26,25 @@ typedef sycl::ext::oneapi::experimental::command_graph<
 typedef sycl::ext::oneapi::experimental::node *node_ptr;
 
 namespace detail {
+
+///  Auxiliary function to query graph nodes.
+/// \param [in] graph A pointer to the command graph.
+/// \param [in] get_func Lambda function for overloading
+/// \param [in] expectednumberOfNodes The number of nodes
+/// to be added
+inline std::vector<sycl::ext::oneapi::experimental::node> get_nodes_impl(
+    std::function<std::vector<sycl::ext::oneapi::experimental::node>()> get_func,
```

Comment:
Can we simplify the parameter?
```suggestion
   const std::vector<sycl::ext::oneapi::experimental::node> &nodes,
```